tools/internal_ci/helper_scripts/prepare_build_linux_rc

```diff
@@ -41,4 +41,29 @@ export DOCKERHUB_ORGANIZATION=grpctesting
 
 git submodule update --init
 
+if [ "${PREPARE_BUILD_INSTALL_DEPS_PYTHON}" == "true" ]
```

Comment:
We currently don't have the "PREPARE_BUILD_INSTALL_*" section for any of the languages for linux and I don't think we should have one.
For vast majority of our linux builds, we rely on all the necessary dependencies being available in the docker images we have. We only use "PREPARE_BUILD_INSTALL_*" on macos and windows, where we can't use docker.
The problem of these per-language setup snippets is that they 1. add complexity 2. take time (= slow down the builds) 3. increase flakiness (unless a lot of extra care is given).

- Building python from source is slow and we really should not do that at the beginning of any of our linux jobs (there must be a better solution)
- Strictly speaking, the name "PREPARE_BUILD_INSTALL_DEPS_PYTHON" doesn't actually fit your use case - since you're not actually installing dependencies for the grpc-python build, but dependencies for our build/test scripts (in this case for the extract_metadata_from_bazel.py). Since any of these scripts might potentially use type annotations (requires python3.6), it also means that you'd eventually need a slow "build python from source" step for almost all of our linux jobs (and I don't think that's acceptable).

Reply:
If customizing the Kokoro image is the practice we want to enforce on Linux jobs, this PR won't make sense.

I do see the conflicts about `PREPARE_BUILD_INSTALL_DEPS_PYTHON`, and open to name changes.

As suggested, we can remove the type annotations to fix the broken tests. I will do that instead and close this PR.

Reply:
Thanks for your understanding!